docs/src/man/extending_ls.md

```diff
@@ -1,3 +1,60 @@
+
+The base callback function ([`LiveServer.file_changed_callback`](@ref)) does only one thing: it sends a signal to the relevant viewers so that page reloads can be triggered.
+You will typically want to re-use `file_changed_callback` or copy its code.
+
+As an example of a custom callback, here is a simple modified callback mechanism which prints `Hello!` before using the base callback function:
+
+```julia
+custom_callback(fp::AbstractString) = (println("Hello!"); file_changed_callback(fp))
+```
+
+A (slightly) more sophisticated customised callback is the one that is used in [`servedocs`](@ref) which checks if the file that was changed is a `.md` file in which case it triggers a pass of `Documenter.jl` before signalling to current viewers.
+
+## Writing your own `FileWatcher`
+
+If you decide to write your own `FileWatcher` type, you will need to meet the API.
+The easier is probably that you look at the code for `SimpleWatcher` and adapt to your need.
+Let's assume for now that you want to define `CustomWatcher <: FileWatcher`.
+
+### Fields
+
+Your `CustomWatcher` will need to have at least the same fields as `SimpleWatcher` i.e.:
+
+* `callback`: the callback function to be triggered upon an event
+* `task`: the asynchronous file watching task
+* `sleeptime`: the time to wait before going over the list of `watchedfiles` to check for events, you won't want this to be less than `0.05`
+* `watchedfiles`: the vector of [`LiveServer.WatchedFile`](@ref)
+* `status`: a symbol that must be set to `:interrupted` upon errors in the file watching task
+
+### Methods
+
+Subsequently, your `CustomWatcher` may redefine some or all of the following methods (those that aren't will use the default method defined for all `FileWatcher`).
+
+* `file_watcher_task!`: this is the loop that goes over the watched files, will check for an event (modifications) and trigger the callback function; this will be the `CustomWatcher.task`. If errors happen in this asynchronous task, the `CustomWatcher.status` should be set to `:interrupted` so that the information can be propagated.
+* `set_callback!`: a helper function to bind a watcher with a callback function.
+* `is_running`: a helper function to check whether `CustomWatcher.task` is done.
+* `start`: start `@async file_watcher_task!`
+* `stop`: stop the `CustomWatcher.task`
+* `is_watched`: check if a file is watched by the watcher.
+
+## Why not use `FileWatching`?
+
+You may be aware of the [`FileWatching`](https://docs.julialang.org/en/v1/stdlib/FileWatching/index.html) module in `Base` and may wonder why we did not just use that one.
+The main reasons we decided not to use it are:
+
+* it triggers **a lot**: where our system only triggers the callback function upon an actual file change (i.e. you modified the file and saved the modification), `FileWatching` is much more sensitive (for instance it will trigger when you _open_ the file),
```

Comment:
That's not entirely true. When I open `index.html` in Atom and just hit `Ctrl-S` repeatedly, the browser updates each time (but I haven't changed a thing in the file).

Reply:
Ok will adjust that it triggers when you press save ;)